query node - GraphQL Playground query templates

ondratra opened the pull request:
This PR overloads dependency on GraphQL Playground to the forked version that features custom query templates. The actual Playground changes are in PR https://github.com/ondratra/graphql-playground/pull/2

ondratra committed:
query node - GraphQL Playground dependency overload I

ondratra commented:
Remaining problems to be solved:
1) improve the way how playground files (e.g. `middleware.js`) files are provided. Currently, files are hosted on my private CDN. So far, I haven't found a way how to serve these files via the Warhog GraphQL server, but there might be a way.
2) add query templates if technically possible: (some are already present)
```
- Get all videos/channels
- Get all videos/channels created after
- Get all videos in channel <id>
- Get all channels owned by member <id>
- Get all videos in category <name>
- Get all videos with license <name>
- Get all videos w/o assets (thumbnail and/or video)
- Get all videos w/o assets (avatar and/or cover)
- Get all videos w/o description/category/language
- etc...
```
3) create `graphql-playground` repository under Joystream organization and move PR from ondratra/graphql-playground#1 there

ondratra committed:
query node - GraphQL Playground dependency overload II

ondratra committed:
query node - GraphQL Playground dependency overload III

mnaamani commented on package.json:
```diff
@@ -37,6 +37,7 @@
     "@polkadot/util": "^6.0.5",
     "@polkadot/util-crypto": "^6.0.5",
     "@polkadot/wasm-crypto": "^4.0.2",
+    "warthog": "https://github.com/ondratra/warthog/releases/download/v2.23.1/warthog-v2.23.1.tgz",
```
Is this published version that depends on your modified `graphql-playground` package?

ondratra replied:
I managed to merge changes to Joystream's fork of Warthog repo. My own repository is no longer required, and https://github.com/Joystream/warthog/releases/download/v2.37.0/joystream-warthog-v2.37.0.tgz is used instead.

mnaamani commented:
This is really nice addition. So I'm seeing the queries, and tried a few. It seems the ones that include `DataObjectOwner` are failing with error below:

```
{
  "error": {
    "errors": [
      {
        "message": "Field \"owner\" of type \"DataObjectOwner!\" must have a selection of subfields. Did you mean \"owner { ... }\"?",
        "locations": [
          {
            "line": 14,
            "column": 7
          }
        ],
        "extensions": {
          "code": "GRAPHQL_VALIDATION_FAILED"
        }
      }
    ]
  }
}
```

ondratra committed:
query node - GraphQL Playground dependency overload IV - models and services adapted

ondratra committed:
query node - playground custom templates ommiting 'owner' property

ondratra commented:
`DataObjectOwner` has known limitations in Sumer, and it's not easily query-able. I intended to omit `owner` property in queries, initially fixed it for `DataObject` entity only. `videos` and `channels` are now fixed as well. There is a built-in feature for omitting some properties (like annoying `createdById` which is useless at the moment), so I just added `owner` to the list.
